app/templates/results.html

```diff
@@ -4,14 +4,29 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Extraction Results</title>
+    <script>
+        function updateCanvas(imageId, textId, canvasId) {
+            const image = document.getElementById(imageId);
+            const text = document.getElementById(textId).value;
+            const canvas = document.getElementById(canvasId);
+            const ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+            ctx.drawImage(image, 0, 0, canvas.width, canvas.height);
+            ctx.font = '20px Arial';
+            ctx.fillStyle = 'red';
+            ctx.fillText(text, 10, 50);
+        }
```

Comment:
What does this do?

Reply:
The JavaScript function `updateCanvas` is designed to dynamically update the content of an HTML5 canvas element based on user input. Here's a breakdown of what each part of the function does:

1. **Function Definition**: `function updateCanvas(imageId, textId, canvasId)` - This function takes three parameters:
   - `imageId`: The ID of the `<img>` element containing the image.
   - `textId`: The ID of the `<input>` element where the user types the text.
   - `canvasId`: The ID of the `<canvas>` element where the image and text are to be drawn.

2. **Getting Elements**:
   - `const image = document.getElementById(imageId);` - Retrieves the image element using its ID.
   - `const text = document.getElementById(textId).value;` - Retrieves the current value (text) from the input field.
   - `const canvas = document.getElementById(canvasId);` - Retrieves the canvas element.

3. **Canvas Context**:
   - `const ctx = canvas.getContext('2d');` - Gets the 2D rendering context for the canvas, which is used for drawing.

4. **Clear Canvas**:
   - `ctx.clearRect(0, 0, canvas.width, canvas.height);` - Clears the entire canvas to ensure that previous drawings do not overlap with the new ones.

5. **Draw Image**:
   - `ctx.drawImage(image, 0, 0, canvas.width, canvas.height);` - Draws the image onto the canvas. The image is positioned at the top-left corner (0,0) and scaled to fit the dimensions of the canvas.

6. **Set Text Style**:
   - `ctx.font = '20px Arial';` - Sets the font size and type for the text to be drawn.
   - `ctx.fillStyle = 'red';` - Sets the color of the text.

7. **Draw Text**:
   - `ctx.fillText(text, 10, 50);` - Draws the text onto the canvas. The text is positioned 10 pixels from the left and 50 pixels from the top of the canvas.

This function is triggered whenever the user types in the text input field associated with an image, allowing for real-time updates of the text overlay on the image within the canvas.